### DISCUSSION TITLE
Hobby Plan Questions

### DISCUSSION THREAD

**Discussion by devstarttips** [vercel/community#1913] [Help] [answered]
### Summary

I had a few questions that I couldn't find the answer to on Vercel's site. Would anyone be able to help with the following questions?:

1. What are the limits with the Hobby hosting plan? How long would I actually be able to host my site?
2. Would my source code/github repository need to be set to Public, or would this work when the repo is Private?
3. Can the Hobby plan be used for Github Enterprise repositories, or just regular Github repos?
4. Is it easy to deploy Flask applications using Vercel?

Thanks in advance for your help!

**Comment by amyegan** [comment, marked as answer]
Great questions! I'm going to try to keep these answers short, but I'll include links to documentation where you can find more details.
1. The Hobby plan limits are listed on our [Limits & Limitations](https://vercel.com/docs/concepts/limits/overview#) page. There isn't a time limit, so you could have your site with us forever.
2. You can [deploy private Git repositories](https://vercel.com/docs/concepts/deployments/git#deploying-private-git-repositories) with Vercel. Public repos are also allowed 🙂 
3. Vercel does not support [connecting GitHub organization repos](https://vercel.com/docs/concepts/limits/overview#connecting-a-project-to-a-git-repository) to Hobby Projects for auto-deployment. You would need a Pro or Enterprise Team for that. But you can [use GitHub Enterprise Server with Vercel](https://vercel.com/guides/can-i-use-github-enterprise-server-with-vercel) with manual deployment via [Vercel CLI](https://vercel.com/docs/cli) or by setting up a [Vercel Deployment GitHub Action](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel)
4. Vercel is a serverless environment, so architecture and project setup is a little different from traditional servers. But I know there are people running Flask projects on Vercel with the [Python Runtime](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/python). We have a Flask [template](https://vercel.com/new/clone?repository-url=https://github.com/vercel/examples/tree/main/python/flask) and [repo](https://github.com/vercel/examples/tree/main/python/flask) if you want to see how it works.

**Reply by devstarttips** [reply]
Thanks for your help!
